pkg/test/framework/components/namespace/namespace.go

```diff
@@ -51,22 +51,22 @@ func ClaimOrFail(t test.Failer, ctx resource.Context, name string) Instance {
 }
 
 // New creates a new Namespace.
-func New(ctx resource.Context, prefix string, inject bool) (i Instance, err error) {
+func New(ctx resource.Context, prefix string, inject bool, labels map[string]string) (i Instance, err error) {
```

Comment:
Perhaps we can wrap prefix, inject, and labels into a struct (e.g.  type metadata struct {...} or type options struct {...})? @ozevren